CryptKeeperZK/crypt-keeper-extension
# feat: download backup notification

## Explanation

This PR adds notification and history track for downloading backup.

Details are below:
- [x] Add notification for download and upload backup
- [x] Track download and upload for history operations
- [x] Add tests for crypto service using property base testing
- [x] Minor renaming

## More Information

Closes #257 

## Screenshots/Screencaps

![image](https://user-images.githubusercontent.com/14254374/236326142-5508e214-bef9-47f3-86d2-2c17246bfc42.png)
![image](https://user-images.githubusercontent.com/14254374/236326047-3aa63b41-b40b-42c3-9f9f-6470468320c7.png)

## Manual Testing Steps

1. Go to advanced settings
2. Backup extension data
3. Enter password and download json file
4. See notification and history operation is written

## Pre-Merge Checklist


> PR template source from [github.com/MetaMask](https://github.com/MetaMask)
